doc/CHANGELOG.rst

```diff
@@ -6,6 +6,9 @@ This changelog *only* contains changes from the *first* pypi release (0.5.4.3) o
 - **Next version - unreleased**
   - java.util.Map conforms to Python abc.Mapping API.
 
+  - float properly follows Java rules for conversion from double.
+    floats outside of range map to inf and -inf.
+
```

Comment:
I think you also added domain boundary checks for integral types. These just used to overflow, now the user will face an exception (ValueError).

Reply:
This is one of the parts that I am unsure about.   I believe the correct behavior is to throw overflow on an automatic conversion and not throw when it is a cast....

```
    i  = 2**34
    test = JClass('jpype.common.Fixture')()
    test.callInt(i)  <= this should throw OverflowError
```

but

```
    i  = 2**34
    test = JClass('jpype.common.Fixture')()
    test.callInt(JInt(i))  <= this should NOT throw OverflowError
```

The reason being that ``test.callInt(JInt(i))`` is equal to ``test.callInt((int)i)`` on the Java side.   Thus if Java doesn't overflow error then Python should not as well.   I don't think I have achieved this correct behavior yet.   

Any thoughts?

Reply:
You're right, that implicit conversion should raise, while explicit shouldn't. But I do not know how to achieve it - sorry.

Reply:
I think that it is just adding a bypass flag on PyJPNumber_new.   But I want to do this as a separate PR rather than part of a huge change ball.